components/motor/gpio/motor_encoder.go

```diff
@@ -442,59 +460,46 @@ func (m *EncodedMotor) goForInternal(ctx context.Context, rpm, revolutions float
 	default:
 	}
 
-	currentPos, err := m.position(ctx, nil)
-	if err != nil {
-		return err
-	}
-
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
 
-	m.state.regulated = true
-	if revolutions == 0 {
-		if m.loop != nil {
-			velVal := math.Abs(rpm * m.ticksPerRotation / 60)
-			// when rev = 0, only velocity is controlled
-			// setPoint is +/- infinity, maxVel is calculated velVal
-			if err := m.updateControlBlock(ctx, math.Inf(int(rpm)), velVal); err != nil {
-				return err
-			}
-		} else {
-			// Moving 0 revolutions is a special value meaning "move forever."
-			oldRpm := m.state.goalRPM
-			m.state.goalRPM = rpm
-			m.state.goalPos = math.Inf(int(rpm))
-			// if we are already moving, let rpmMonitor deal with setPower
-			if math.Abs(oldRpm) > 0.001 && m.state.direction == m.directionMovingInLock() {
-				return nil
-			}
-			// if moving from stop, start at 10% power
-			if err := m.setPower(ctx, m.state.direction*0.1, true); err != nil {
-				return err
-			}
+	switch {
+	case m.loop != nil && goalPos == 0:
+		velVal := math.Abs(rpm * m.ticksPerRotation / 60)
+		// when rev = 0, only velocity is controlled
+		// setPoint is +/- infinity, maxVel is calculated velVal
+		if err := m.updateControlBlock(ctx, math.Inf(int(rpm)), velVal); err != nil {
+			return err
 		}
-		return nil
-	}
-
-	goalPos := (math.Abs(revolutions) * m.ticksPerRotation * m.state.direction) + currentPos
-
-	m.state.goalRPM = math.Abs(rpm) * m.state.direction
-	m.state.goalPos = goalPos
-
-	if m.loop != nil {
+	case m.loop != nil && goalPos != 0:
 		velVal := math.Abs(rpm * m.ticksPerRotation / 60)
 		// when rev is not 0, velocity and position are controlled
 		// setPoint is goalPos, maxVel is calculated velVal
 		if err := m.updateControlBlock(ctx, goalPos, velVal); err != nil {
 			return err
 		}
-	} else {
-		startingPwr := 0.1 * m.state.direction
-		err = m.setPower(ctx, startingPwr, true)
+	case m.loop == nil && goalPos == 0:
```

Comment:
I think in order for these cases to work we need to set at least `state.goalRPM` and `state.goalPos` in `rpmMonitor` and maybe also `state.direction`. so I'm not sure there's a point in passing these values into `rpmMonitor` instead of just setting them beforehand

Reply:
or I guess I'm not sure/don't understand if there's a benefit to passing these values to rpmMonitor